src/Nethermind/Nethermind.Merge.AuRa/Shutter/ShutterTxLoader.cs

```diff
@@ -171,36 +208,60 @@ internal Transaction[] DecryptSequencedTransactions(List<SequencedTransaction> s
 
     private IEnumerable<SequencedTransaction> GetNextTransactions(ulong eon, ulong txPointer, long headBlockNumber)
     {
-        IEnumerable<ISequencerContract.TransactionSubmitted> events = _sequencerContract.GetEvents(eon, txPointer, headBlockNumber);
-        if (_logger.IsDebug) _logger.Debug($"Found {events.Count()} events in Shutter sequencer contract.");
-
-        UInt256 totalGas = 0;
-        int index = 0;
-
-        foreach (ISequencerContract.TransactionSubmitted e in events)
+        lock (_transactionSubmittedEvents)
         {
-            if (totalGas + e.GasLimit > _encryptedGasLimit)
+            if (_firstLoad)
             {
-                if (_logger.IsDebug) _logger.Debug("Shutter gas limit reached.");
-                yield break;
+                _transactionSubmittedEvents = _sequencerContract.GetEvents(eon, txPointer, headBlockNumber).ToList();
+                _txPointer = _transactionSubmittedEvents.Count == 0 ? txPointer : (_transactionSubmittedEvents.Last().TxIndex + 1);
+                _firstLoad = false;
+                // todo: make debug
+                if (_logger.IsInfo) _logger.Info($"Found {_transactionSubmittedEvents.Count} Shutter events from scanning logs up to block {headBlockNumber}, local tx pointer is {_txPointer}.");
             }
+            else
+            {
+                // todo: make debug
+                if (_logger.IsInfo) _logger.Info($"Found {_transactionSubmittedEvents.Count} Shutter events from recent blocks up to {headBlockNumber}, local tx pointer is {_txPointer}.");
+            }
+            List<ISequencerContract.TransactionSubmitted> events = _transactionSubmittedEvents.ToList();
 
-            byte[] identityPreimage = new byte[52];
-            e.IdentityPrefix.AsSpan().CopyTo(identityPreimage.AsSpan());
-            e.Sender.Bytes.CopyTo(identityPreimage.AsSpan()[32..]);
+            UInt256 totalGas = 0;
+            int index = 0;
 
-            SequencedTransaction sequencedTransaction = new()
+            foreach (ISequencerContract.TransactionSubmitted e in events)
```

Comment:
if you do a `for` loop, you can change the iterated collection, just be sure to not advance index if you remove an item